citest/gcp_testing/gcp_agent.py

```diff
@@ -0,0 +1,287 @@
+    method_spec = info.get('methods', {}).get(method, None)
+    if method_spec is None:
+      raise KeyError('Unknown method={0} on resource={1}'
+                     .format(method, resource_type))
+    parameters = method_spec.get('parameters', {})
+    result = {key: value for key, value in kwargs.items() if value is not None}
+    resource_id_param = None
+    if resource_id is not None:
+      resource_id_param = method_spec['parameterOrder'][-1]
+      result[resource_id_param] = resource_id
+
+    missing = []
+    for key, key_spec in parameters.items():
+      if key not in kwargs and key != resource_id_param:
+        def_value = self.__default_variables.get(key, None)
+        if def_value is not None:
+          result[key] = def_value
+        elif key_spec.get('required', False):
+          missing.append(key)
+
+    if missing:
+      raise ValueError('"{0}()" is missing required parameters: {1}'
+                       .format(method, missing))
+    return result
+
+  def get_resource(self, resource_type, resource_id=None, **kwargs):
+    """Get instance metadata details.
+
+    Args:
+      resource_type: [string] The type of the resource instance to get.
+      resource_id: [string] The id of the resource instance to get.
+      kwargs: [kwargs] Additional parameters may be required depending
+         on the resource type (such as zone, etc).
+    """
+    return self.invoke_resource(
+        'get', resource_type=resource_type, resource_id=resource_id, **kwargs)
+
+  def invoke_resource(self, method, resource_type, resource_id=None, **kwargs):
+    """Invoke a method on a resource type or instance.
+
+    Args:
+      method: [string] The operation to perform as named under the
+          |resource_type| in the discovery document.
+      resource_type: [string] The type of the resource instance to operate on.
+      resource_id: [string] The id of the resource instance, or None to operate
+        on the resource type or collection.
+      kwargs: [kwargs] Additional parameters may be required depending
+         on the resource type and method.
+    """
+    variables = self.resource_method_to_variables(
+        method, resource_type, resource_id=resource_id, **kwargs)
+
+    resource_obj = vars(self.__service).get(resource_type, None)
+    if resource_obj is None:
+      raise KeyError('Unknown {api} Resource type={type}'
+                     .format(api=self.__api_title, type=resource_type))
+
+    JournalLogger.journal_or_log(
+        'Requesting {type} {method} {vars}'.format(
+            type=resource_type, method=method, vars=variables),
+        _module=self.logger.name,
+        _context='request')
+
+    request = getattr(resource_obj(), method)(**variables)
+    response = request.execute()
+    JournalLogger.journal_or_log(
+        json.JSONEncoder(
+            encoding='utf-8', separators=(',', ': ')).encode(response),
+        _module=self.logger.name, _context='response',
```

Comment:
The ".encode" call tacked on to the result of JSONEncoder might be throwing you off. 
pylint is happy.